src/main.rs

```diff
@@ -1,30 +1,67 @@
+use std::env;
+use std::process;
+use std::{cell::RefCell, str::FromStr};
+
+use getopts::Options;
 use hprof::Profiler;
 use libgir::{self as gir, Config, Library, WorkMode};
-use std::{cell::RefCell, str::FromStr};
 
-static USAGE: &str = "
-Usage: gir [options] [<library> <version>]
-       gir (-h | --help)
-
-Options:
-    -h, --help              Show this message.
-    -c CONFIG               Config file path (default: Gir.toml)
-    -d GIRSPATH             Directory for girs
-    -m MODE                 Work mode: doc, normal, sys or not_bound
-    -o PATH                 Target path
-    --doc-target-path PATH  Doc target path
-    -b, --make-backup       Make backup before generating
-    -s, --stats             Show statistics
-";
+fn print_usage(program: &str, opts: Options) {
+    let brief = format!(
+        "Usage: {program} [options] [<library> <version>]
+       {program} (-h | --help)",
+        program = program
+    );
+    print!("{}", opts.usage(&brief));
+}
+
+trait OptionStr {
+    fn as_str_ref(&self) -> Option<&str>;
+}
+
+impl<S: AsRef<str>> OptionStr for Option<S> {
+    fn as_str_ref(&self) -> Option<&str> {
+        self.as_ref().map(|string| string.as_ref())
+    }
+}
 
 fn build_config() -> Result<Config, String> {
-    let args = match docopt::Docopt::new(USAGE).and_then(|dopt| dopt.parse()) {
-        Ok(args) => args,
+    let args: Vec<_> = env::args().collect();
+    let program = args[0].clone();
+
+    let mut options = Options::new();
+    options.optopt(
+        "c",
+        "config",
+        "Config file path (default: Gir.toml)",
+        "CONFIG",
+    );
+    options.optflag("h", "help", "Show this message");
```

Comment:
Compared to docopt or clap this is quite ugly :)

Reply:
Yep but it's just options and quite a small amount of code so who cares. :)